framework/src/task_manager/task_manager_core.c

```diff
@@ -556,59 +562,154 @@ int taskmgr_get_handle_by_pid(int pid)
 	return TM_UNREGISTERED_APP;
 }
 
+static int taskmgr_get_tm_broad_msg(int msg)
+{
+	int tm_broadcast_msg_idx;
+	
+	for (tm_broadcast_msg_idx = 0; tm_broadcast_msg_idx < CONFIG_TASK_MANAGER_MAX_TASKS; tm_broadcast_msg_idx++) {
+		if (tm_broadcast_msg[tm_broadcast_msg_idx].input_msg == msg) {
+			return tm_broadcast_msg[tm_broadcast_msg_idx].tm_broad_msg;
+		}
+	}
+	return TM_INVALID_BROADCAST_MSG;
+}
+
+static tm_broadcast_t *taskmgr_search_broadcast_info(int msg, int handle)
+{
+	int tm_broad_msg;
+	sq_entry_t curr;
+
+	tm_broad_msg = taskmgr_get_tm_broad_msg(msg);
+	curr.flink = TM_BROADCAST_INFO_LIST(handle).head;
+	while (curr.flink != NULL) {
+		if (((tm_broadcast_t *)curr.flink)->msg == tm_broad_msg) {
+			return ((tm_broadcast_t *)curr.flink);
+		} else {
+			curr.flink = ((FAR sq_entry_t *)curr.flink)->flink;
+		}
+	}
+	return NULL;
+}
+
 static void taskmgr_broadcast(int msg)
 {
 	int handle;
 	int fd;
 	int ret;
 	union sigval msg_broad;
+	tm_broadcast_t *broadcast_info;
 
 	fd = taskmgr_get_drvfd();
 	if (fd < 0) {
 		return;
 	}
-	msg_broad.sival_int = msg;
+
 	for (handle = 0; handle < CONFIG_TASK_MANAGER_MAX_TASKS; handle++) {
 		if (TM_LIST_ADDR(handle) != NULL) {
 			ret = taskmgr_get_task_state(handle);
 			if (ret == TM_APP_STATE_STOP || ret == TM_APP_STATE_UNREGISTERED) {
 				continue;
 			}
-			ret = taskmgr_check_msg_mask(msg, handle);
-			if (ret != OK) {
+			ret = taskmgr_get_tm_broad_msg(msg);
```

Comment:
checking the return of taskmgr_get_tm_broad_msg is not needed?

Reply:
Done